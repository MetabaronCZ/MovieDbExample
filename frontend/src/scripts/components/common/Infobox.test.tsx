import { getByText } from '@testing-library/dom';
import { describe, expect, it } from '@jest/globals';

import { Infobox } from './Infobox';
import { TestComponentWrapper, waitForComponent } from 'test-utils/component';

describe('components/common/Infobox', () => {
  it('should render properly', async () => {
    const { container } = await waitForComponent(<Infobox>CONTENT</Infobox>, {
      wrapper: TestComponentWrapper,
    });
    expect(container).toMatchSnapshot();

    const content = getByText(container, 'CONTENT');
    expect(content).toBeInTheDocument();
  });

  it('should render variations', async () => {
    const { container: info } = await waitForComponent(
      <Infobox type="info">INFO</Infobox>,
      { wrapper: TestComponentWrapper },
    );
    expect(info).toMatchSnapshot();

    const { container: error } = await waitForComponent(
      <Infobox type="error">ERROR</Infobox>,
      { wrapper: TestComponentWrapper },
    );
    expect(error).toMatchSnapshot();

    const { container: success } = await waitForComponent(
      <Infobox type="success">SUCCESS</Infobox>,
      { wrapper: TestComponentWrapper },
    );
    expect(success).toMatchSnapshot();
  });
});
