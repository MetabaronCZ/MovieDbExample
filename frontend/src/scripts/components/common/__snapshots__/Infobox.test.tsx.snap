// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/common/Infobox should render properly 1`] = `
.c0 {
  font-family: Tahoma,Verdana,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
  color: #002d56;
}

.c1 {
  padding: 8px 16px;
  color: #002d56;
  background: #fff;
  border: 1px solid #002d56;
  border-color: #002d56;
  box-shadow: 1px 1px 4px #dae0e3;
}

<div>
  <p
    class="c0 c1"
  >
    CONTENT
  </p>
</div>
`;

exports[`components/common/Infobox should render variations 1`] = `
.c0 {
  font-family: Tahoma,Verdana,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
  color: #002d56;
}

.c1 {
  padding: 8px 16px;
  color: #002d56;
  background: #fff;
  border: 1px solid #002d56;
  border-color: #002d56;
  box-shadow: 1px 1px 4px #dae0e3;
}

<div>
  <p
    class="c0 c1"
  >
    INFO
  </p>
</div>
`;

exports[`components/common/Infobox should render variations 2`] = `
.c0 {
  font-family: Tahoma,Verdana,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
  color: #002d56;
}

.c1 {
  padding: 8px 16px;
  color: #f00;
  background: #fff;
  border: 1px solid #002d56;
  border-color: #f00;
  box-shadow: 1px 1px 4px #dae0e3;
}

<div>
  <p
    class="c0 c1"
  >
    ERROR
  </p>
</div>
`;

exports[`components/common/Infobox should render variations 3`] = `
.c0 {
  font-family: Tahoma,Verdana,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
  color: #002d56;
}

.c1 {
  padding: 8px 16px;
  color: #0a0;
  background: #fff;
  border: 1px solid #002d56;
  border-color: #0a0;
  box-shadow: 1px 1px 4px #dae0e3;
}

<div>
  <p
    class="c0 c1"
  >
    SUCCESS
  </p>
</div>
`;
